merkledag/cidversion.go

```diff
@@ -0,0 +1,26 @@
+package merkledag
+
+import (
+	"fmt"
+)
+
+// CidVersion is a a cid version number that is guaranteed to be valid
+type CidVersion struct {
```

Comment:
I think you went the way you did because as a way of preventing creation of different CID versions but IMO it should be done with helper function when CID version is from User Interface.